src/Utilities/Compiler/Options/SymbolNamesOption.cs

```diff
@@ -101,13 +122,27 @@ public override bool Equals(object obj)
         public bool Equals(SymbolNamesOption? other)
         {
             return other != null &&
-                _names.SetEquals(other._names) &&
-                _symbols.SetEquals(other._symbols);
+                _names.Count == other._names.Count &&
+                _symbols.Count == other._symbols.Count &&
+                _names.Keys.All(key => other._names.ContainsKey(key) && string.Equals(_names[key], other._names[key], StringComparison.Ordinal)) &&
```

Comment:
I thought we had some helper/extension method for dictionary equality. Can you check? If not, can you extract this into one?